### PR TITLE
Add full support for key-based authentication

### PR DESCRIPTION
**Pull request by ogajduse**
This PR adds support for key-based authentication to the `broker.hosts.Host`s.

### CODE REVIEW

**Comment by JacobCallahan** [comment]
Can you outline the testing scenarios you've tried with this change?

Also, it might be good to add some kind of file existence validation before attempting auth (falling back to username/pass)

**Comment by ogajduse** [comment]
> Can you outline the testing scenarios you've tried with this change?
> 
> Also, it might be good to add some kind of file existence validation before attempting auth (falling back to username/pass)

I added the check for file existence. An exception is thrown if the file does not exist - no implicit fallback to username/pass.

I will add test scenarios tomorrow.

**Comment by ogajduse** [comment]
[asciicast](https://asciinema.org/a/xtHt3YSuM495DHOhH8BKhHv6J)

File: broker_settings.yaml.example [hunk 1]
```diff
@@ -5,6 +5,7 @@ inventory_file: "inventory.yaml"
 host_username: "root"
 host_password: "<password>"
 host_ssh_port: "<port>"
+host_ssh_key_filename: "<path-to-the-ssh-key>"
```
Comment (JacobCallahan) [inline broker_settings.yaml.example:8]:
```suggestion
host_ssh_key_filename: "</path/to/the/ssh-key>"
```

**Review by JacobCallahan — APPROVED** [review]

**Review by mshriver — APPROVED** [review]

**Review by JacobCallahan — APPROVED** [review]
ACK, thanks for the addition!

**Pull request merged**